xstate-chatbot/nodejs/src/machine/hospital-details.js

```diff
@@ -0,0 +1,1068 @@
+                  context.slots.hospital.deaths_covid_patients_without_venti_l3 = event.data.data[0].deaths_covid_patients_without_venti_l3;
+                  context.slots.hospital.no_cases_on_intubated_invasive_venti_l3 = event.data.data[0].no_cases_on_intubated_invasive_venti_l3;
+                  context.slots.hospital.discharged_covid_patients_with_venti_l3 = event.data.data[0].discharged_covid_patients_with_venti_l3;
+                  context.slots.hospital.deaths_covid_patients_with_venti_l3 = event.data.data[0].deaths_covid_patients_with_venti_l3;
+                  context.slots.previoushospitaldata.bed_capacity_icu_L3 = event.data.data[0].bed_capacity_icu_L3;
+                  context.slots.previoushospitaldata.bed_capacity_L3 = event.data.data[0].bed_capacity_L3;
+                  context.slots.previoushospitaldata.no_cases_on_icu_niv_without_venti_l3 = event.data.data[0].no_cases_on_icu_niv_without_venti_l3;
+                  context.slots.previoushospitaldata.discharged_covid_patients_without_venti_l3 = event.data.data[0].discharged_covid_patients_without_venti_l3;
+                  context.slots.previoushospitaldata.deaths_covid_patients_without_venti_l3 = event.data.data[0].deaths_covid_patients_without_venti_l3;
+                  context.slots.previoushospitaldata.no_cases_on_intubated_invasive_venti_l3 = event.data.data[0].no_cases_on_intubated_invasive_venti_l3;
+                  context.slots.previoushospitaldata.discharged_covid_patients_with_venti_l3 = event.data.data[0].discharged_covid_patients_with_venti_l3;
+                  context.slots.previoushospitaldata.deaths_covid_patients_with_venti_l3 = event.data.data[0].deaths_covid_patients_with_venti_l3;
+                  dialog.sendMessage(context, dialog.get_message(messages.nodalOfficer.prompt, context.user.locale));
+                }),
+                target: '#hospitalTypeId',
+              },
+              {
+                target: 'error',
+              },
+            ],
+          },
+        },
+        error: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.noUserFound, context.user.locale), false);
+          }),
+          always: 'prompt',
+        },
+      },
+    },//Nodal Officer 
+    hospitalTypeId: {
+      id: 'hospitalTypeId',
+      initial: 'process',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+          }),
+          on: {
+            USER_MESSAGE: 'process',
+          },
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            //context.slots.hospital.mobileNumber=context.user.mobileNumber;
+            if (context.slots.previoushospitaldata.hospital_level_Id == '1') {
+              let message = dialog.get_message(messages.L2L3hospitalDetails, context.user.locale);
+              dialog.sendMessage(context, message);
```

Comment:
If you are asking something to the user then send the message as part of the prompt state.
If you are just providing information to the user make it a separate state. Refer to covid-info.js

Reply:
message are different among L2,L3 and L2L3 nodal officer so I mentioned different different messages for all three under prompt--onEntry

Reply:
Do we need to send messages in prompt as well as process?
Is it working fine in your local machine. I think this would be sending multiple duplicate messages.

Reply:
mesage required in process only as per requirement

Reply:
Even when there is dialog.sendMessage in both prompt and process, is it working properly in your local?

Reply:
previously the code was working because of this only [initial: 'process'] and now dialog.sendMessage is there in process only (prompt is removed)